vnext/ReactUWP/Views/ShadowNodeBase.cpp

```diff
@@ -9,14 +9,26 @@
 #include <Views/ShadowNodeBase.h>
 #include <Views/ViewManagerBase.h>
 #include <Views/ExpressionAnimationStore.h>
+#include "Views/KeyboardEventHandler.h"
 #include <winrt/Windows.UI.Composition.h>
 #include <WindowsNumerics.h>
 
+using namespace std::placeholders;
+
 namespace react { namespace uwp {
 
 ShadowNodeBase::ShadowNodeBase()
   : m_view(nullptr)
 {
+  m_previewKeyboardEventHandler = make_unique<PreviewKeyboardEventHandler>(
+    std::bind(&ShadowNodeBase::KeyboardEventHandledHandler, this, KeyboardEventPhase::PreviewKeyDown, _1, _2),
+    std::bind(&ShadowNodeBase::KeyboardEventHandledHandler, this, KeyboardEventPhase::PreviewKeyUp, _1, _2)
+    );
+
+  m_keyboardEventHandler = make_unique<KeyboardEventHandler>(
+    std::bind(&ShadowNodeBase::KeyboardEventHandledHandler, this, KeyboardEventPhase::KeyDown, _1, _2),
+    std::bind(&ShadowNodeBase::KeyboardEventHandledHandler, this, KeyboardEventPhase::KeyUp, _1, _2)
+    );
 }
```

Comment:
This seems a little unnecessarily expensive to have on every shadownode, 2 more heap allocations that will be rarely used?  either delay create or don't heap alloc just have KeyboardEventHandler member instead of a std::unique<KeyboardEventHandler> ?

Reply:
Good suggestion, will do it

---
In reply to: [295397004](https://github.com/microsoft/react-native-windows/pull/2623#discussion_r295397004) [](ancestors = 295397004)